docs/increase-performance.rst

```diff
@@ -0,0 +1,67 @@
+.. _increase-performance:
+
+==============================================================
+Increase Locust's performance with a faster HTTP client
+==============================================================
+
+Locust's default HTTP client uses `python-requests <http://www.python-requests.org/>`_. 
+The reason for this is that requests is a very well-maintained python package, that 
+provides a really nice API, that many python developers are familiar with. Therefore, 
+in many cases, we recommend that you use the default :py:class:`HttpLocust <locust.core.HttpLocust>` 
+which uses requests. However, if your're planning to run really large scale scale tests, 
+Locust comes with an alternative HTTP client, 
+:py:class:`FastHttpLocust <locust.contrib.fasthttp.FastHttpLocust>` which 
+uses `geventhttpclient <https://github.com/gwik/geventhttpclient/>`_ instead of requests.
+This client is significantly faster, and we've seen 5x-6x performance increases for 
+HTTP-requests.
+
+
+Known limitations
+=================
+
+* :py:class:`FastHttpLocust <locust.contrib.fasthttp.FastHttpLocust>` does not 
+  support the :ref:`catch-response <catch-response>` argument.
+* Basic auth is currently not supported.
+
+
+How to use FastHttpLocust
+===========================
+
+First, you need to install the geventhttplocust python package::
+
+    pip install geventhttpclient
```

Comment:
Do you think we should skip this step and just add `geventhttpclient` as a required dependency in setup.py (rather than only installing it in `tests_require`)?  That way we could also pin a version number.

Reply:
Yeah, I think that's a good idea if we can either get the maintainers of geventhttpclient to upload python wheels (for linux, osx & windows), or worst-case create a separate PyPI repository where we upload wheels for it ourselves. But if we do the job of fixing an automated way of building wheels, I see no reason why they wouldn't accept it :).

Without wheels it will otherwise become much harder to install Locust (especially on windows), since it'll require that one has a C compiler and build environment set up.

Reply:
I've opened an issue at the geventhttpclient project asking about this: https://github.com/gwik/geventhttpclient/issues/93

Reply:
And now a pull-request as well: https://github.com/gwik/geventhttpclient/pull/94

Reply:
Per https://github.com/gwik/geventhttpclient/issues/93#issuecomment-353781687 - maybe we  should consider adding a `locustio/geventhttpclient` fork and build our own wheels from that? Or we could check the validity of the wheels built at:  https://pypi.python.org/pypi/geventhttpclient-wheels/ and use those?  The latest pypi push was on 12-23 and seems to reflect the latest `geventhttpclient` master.

Reply:
The geventhttpclient-wheels is my package :). I created it while I was putting together this PR to the geventhttpclient project: https://github.com/gwik/geventhttpclient/pull/94

Hopefully the owner will show up and merge it! Otherwise we'll have to keep a fork which can automatically upload wheels.

Reply:
Could also make it an extras package so it's `pip install locustio[geventhttp]` and not installed all the time.